src/org/sosy_lab/java_smt/api/PropagatorBackend.java

```diff
@@ -0,0 +1,100 @@
+// This file is part of JavaSMT,
+// an API wrapper for a collection of SMT solvers:
+// https://github.com/sosy-lab/java-smt
+//
+// SPDX-FileCopyrightText: 2024 Dirk Beyer <https://www.sosy-lab.org>
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package org.sosy_lab.java_smt.api;
+
+import java.util.Optional;
+
+/**
+ * The PropagatorBackend class is used by {@link UserPropagator} to implement custom user
+ * propagators.
+ * It contains functions to interact with the SAT/SMT core during solving, for example, it provides
+ * the ability to propagate conflicts and to influence the decision-making.
+ */
+public interface PropagatorBackend {
```

Comment:
This class is for internal usage only. A user does not need to implement it himself, but JavaSMT internally provides solver-specific implementation for it. Correct?

Reply:
Yes, it is implemented internally only (and could me marked `sealed` in new Java Versions).
The user only interacts with this backend via this theory solver.